lib/device/device.dart

```diff
@@ -51,172 +44,53 @@ class DeviceScreen extends Screen {
         ])
     ]);
 
-    _rebuildTogglesDiv();
-  }
-
-  void _handleConnectionStart(VmServiceWrapper service) {
-    extensionTracker = new ExtensionTracker(service);
-    extensionTracker.start();
-
-    extensionTracker.onChange.listen((_) {
-      framesChartStateMixin.setState(() {
-        if (extensionTracker.hasIsolateTargets && !visible) {
-          visible = true;
-        }
-
-        _rebuildTogglesDiv();
-      });
-    });
-
-    deviceStatus.element.text =
-        '${serviceManager.vm.targetCPU} ${serviceManager.vm.architectureBits}-bit';
-  }
-
-  void _handleConnectionStop(dynamic event) {
-    extensionTracker?.stop();
-
-    deviceStatus.element.text = '';
+    _buildTogglesDiv();
   }
 
   @override
   HelpInfo get helpInfo => null;
 
-  void _rebuildTogglesDiv() {
-    if (togglesDiv == null || extensionTracker == null) {
+  void _buildTogglesDiv() {
+    if (togglesDiv == null) {
       return;
     }
 
-    togglesDiv.clear();
-
-    _createBoolToggle('ext.flutter.debugPaint');
-    _createBoolToggle('ext.flutter.debugPaintBaselinesEnabled');
-    _createBoolToggle('ext.flutter.repaintRainbow');
-    _createBoolToggle('ext.flutter.showPerformanceOverlay');
-    _createBoolToggle('ext.flutter.debugAllowBanner');
-    _createBoolToggle('ext.flutter.profileWidgetBuilds');
+    _createBoolToggle(extensions.debugPaint);
+    _createBoolToggle(extensions.debugPaintBaselines);
+    _createBoolToggle(extensions.repaintRainbow);
+    _createBoolToggle(extensions.performanceOverlay);
+    _createBoolToggle(extensions.debugAllowBanner);
+    _createBoolToggle(extensions.profileWidgetBuilds);
   }
 
   void _createBoolToggle(String rpc) {
-    if (!extensionTracker.extensionToIsolatesMap.containsKey(rpc)) {
-      return;
-    }
-
-    CoreElement input;
-
-    togglesDiv.add(div(c: 'form-checkbox')
-      ..add(new CoreElement('label')
-        ..add(<CoreElement>[
-          input = new CoreElement('input')..setAttribute('type', 'checkbox'),
-          span(text: rpc),
-        ])));
-
-    if (boolValues.containsKey(rpc)) {
-      input.toggleAttribute('checked', boolValues[rpc]);
-    } else {
-      extensionTracker.callBoolExtensionMethod(rpc).then((bool value) {
-        input.toggleAttribute('checked', value);
-        boolValues[rpc] = value;
-      });
-    }
-
-    input.element.onChange.listen((_) {
-      final html.InputElement e = input.element;
-      boolValues[rpc] = e.checked;
-      extensionTracker.setBoolExtensionMethod(rpc, e.checked);
-    });
-  }
-}
-
-class ExtensionTracker {
-  ExtensionTracker(this.service) {
-    service.onIsolateEvent.listen((Event e) {
-      if (e.kind == 'ServiceExtensionAdded') {
-        _registerRpcForIsolate(e.extensionRPC, e.isolate);
+    serviceManager.serviceExtensionManager.hasServiceExtension(rpc,
```

Comment:
Nit: it would probably be better to make non-available extensions show up as disabled instead of adding and removing checkboxes which could look visually distracting.

Reply:
done